backend/app/services/dingtalk/docs_service.py

```
@@ -0,0 +1,548 @@
+                        "Accept": "application/json",
+                    },
+                )
+                response.raise_for_status()
+                result = response.json()
+        except httpx.HTTPStatusError as e:
+            logger.error(f"HTTP error from DingTalk MCP: {e.response.status_code}")
+            logger.error(f"Response body: {e.response.text}")
+            if e.response.status_code in AUTH_ERROR_CODES:
+                raise ValueError(
+                    "DingTalk authentication failed. Please check your MCP configuration."
+                )
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)